### PR TITLE
 [TRACING-4077] Add documentation for kubernetes cluster receiver

### PR DESCRIPTION
**Pull request by rubenvp8510**
Version(s):

Issue: https://issues.redhat.com/browse/TRACING-3965

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/otel-collector-components.adoc [hunk 1]
```diff
@@ -199,6 +199,109 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
...
+metadata:
+  labels:
+    app: otelcontribcol
+  name: otelcontribcol
+----
+
+.Kubernetes cluster receiver service account RBAC
+[source,yaml]
+----
+piVersion: rbac.authorization.k8s.io/v1
```
Comment (IshwarKanse) [inline modules/otel-collector-components.adoc:240]: Missed the a in apiVersion. We should also add the required ClusterRoleBinding to bind the created role to the SA.

File: modules/otel-collector-components.adoc [hunk 2]
```diff
@@ -199,6 +199,109 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
...
+[id="k8scluster-receiver_{context}"]
+=== Kubernetes cluster receiver
+
+The Kubernetes cluster receiver is responsible for gathering cluster-level metrics and entity events from the Kubernetes API server. It utilizes the K8s API to stay informed about updates. Authentication for this receiver is exclusively supported through service accounts.
+
+.OpenTelemetry Collector custom resource with the enabled Kubernetes cluster receiver
+[source,yaml]
+----
+  receivers:
+    k8s_cluster:
```
Comment (IshwarKanse) [inline modules/otel-collector-components.adoc:211]: Support for collecting OpenShift specific metrics is also available which needs to be documented. https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sclusterreceiver#openshift

File: modules/otel-collector-components.adoc (hunk 1 above)
Comment (rubenvp8510) [reply modules/otel-collector-components.adoc]: Done

File: modules/otel-collector-components.adoc (hunk 2 above)
Comment (rubenvp8510) [reply modules/otel-collector-components.adoc:211]: Done

**Comment by rubenvp8510** [comment]
@IshwarKanse @max-cx I've already address the comments!

**Review by IshwarKanse — APPROVED** [review]

File: modules/otel-collector-components.adoc (hunk 2 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:211]: @IshwarKanse, please ping me when you get around to verifying and approving Ruben's update.

Comment (max-cx) [reply modules/otel-collector-components.adoc:211]: Thank you both.

**Comment by max-cx** [comment]
@rubenvp8510, you're welcome to close this PR because I opened https://github.com/openshift/openshift-docs/pull/75604.

**Pull request closed without merging**